benchmark/buffers/buffer-compare-offset.js

```diff
@@ -0,0 +1,50 @@
+'use strict';
+const common = require('../common.js');
+const v8 = require('v8');
+
+const bench = common.createBenchmark(main, {
+  method: ['offset', 'slice'],
+  size: [16, 512, 1024, 4096, 16386],
+  millions: [1]
+});
+
+function compareUsingSlice(b0, b1, len, iter) {
+
+  // Force optimization before starting the benchmark
+  Buffer.compare(b0.slice(1, len), b1.slice(1, len));
+  v8.setFlagsFromString('--allow_natives_syntax');
+  eval('%OptimizeFunctionOnNextCall(Buffer.compare)');
+  eval('%OptimizeFunctionOnNextCall(b0.slice)');
+  eval('%OptimizeFunctionOnNextCall(b1.slice)');
+  Buffer.compare(b0.slice(1, len), b1.slice(1, len));
+  var i;
+  bench.start();
+  for (i = 0; i < iter; i++)
+    Buffer.compare(b0.slice(1, len), b1.slice(1, len));
```

Comment:
Mind throwing this into it's own function? Easier to view optimized output when run from own function.